sdk-cpp/src/do_download.cpp

```diff
@@ -84,78 +165,65 @@ void download::start_and_wait_until_completion(const std::atomic_bool& isCancell
         {
             pollTime += 500ms;
         }
-        status = get_status();
+        RETURN_IF_FAILED(get_status_nothrow(status));
         timedOut = std::chrono::system_clock::now() >= endTime;
     } while ((status.state() == download_state::created || status.state() == download_state::transferring || status.is_transient_error())
         && !timedOut);
 
     if (status.state() == download_state::transferred)
     {
-        finalize();
+        RETURN_IF_FAILED(finalize_nothrow());
     }
     else
     {
-        abort();
+        RETURN_IF_FAILED(abort_nothrow());
         if (isCancelled)
         {
-            msdod::ThrowException(std::errc::operation_canceled);
+            return DO_ERROR_FROM_STD_ERROR(std::errc::operation_canceled);
```

Comment:
>DO_ERROR_FROM_STD_ERROR [](http://example.com/codeflow?start=19&length=23)

When we return std::error_code instead, we don't have to perform these conversions. Tests will be happy too.